ui/plugins/bookmarks/src/components/bookmarks-page.tsx

```diff
@@ -52,26 +54,23 @@ const BookmarksPage: React.FC<BookmarksPageProps> = props => {
     ev.preventDefault();
   };
 
-  const redirectToPost =
-    (navigateToUrl: (path: string) => void) =>
-    (details: {
-      authorEthAddress: string;
-      entryId: string;
-      replyTo: {
-        authorEthAddress: string;
-        entryId: string;
-      } | null;
-    }) => {
-      const { entryId, replyTo } = details;
-      let url = `/social-app/post/${entryId}`;
-      if (replyTo && replyTo.entryId) {
-        // handle the reply
-        url = `/social-app/post/${replyTo.entryId}`;
-      }
-      navigateToUrl(url);
-    };
-
-  const handleNavigateToPost = redirectToPost(props.singleSpa.navigateToUrl);
+  const handleNavigation = (itemType: ItemTypes, details: IContentClickDetails) => {
+    let url;
+    switch (itemType) {
+      case ItemTypes.PROFILE:
+        url = `/profile/${details.entryId}`;
```

Comment:
all the links are built with `details.entryId` ? O_O

Reply:
Uhhhhm 😅 It is misdirecting, but it's the same pattern used elsewhere in the repo. I could change it to just ie. id everywhere?

Reply:
It's a generic handler for multiple types of redirects, and unless we refactor all of it I wouldn't change it now

Reply:
we should create a todo to refactor this ;/